### PR TITLE
locale.c: Refactor internal debugging function

### PR DESCRIPTION
**Pull request by khwilliamson**
setlocale_debug_string() variants  now use Perl_form, a function I
didn't know existed when I originally wrote this code.

### CODE REVIEW

File: locale.c [hunk 1]
```diff
@@ -6471,56 +6473,66 @@ Perl_sync_locale()
...
+                     locale_quote, locale, locale_quote,
+                     retval_quote, retval, retval_quote);
 }
 
 #endif
 
 void
 Perl_thread_locale_init()
 {
-    /* Called from a thread on startup*/
```
Comment (bram-perl) [inline locale.c]: Why the removal of that line? Seems unrelated to the commit/PR

File: locale.c [hunk 2]
```diff
@@ -6471,56 +6473,66 @@ Perl_sync_locale()
...
     }
 
-    assert(strlen(ret) < sizeof(ret));
+#  ifdef USE_LOCALE_THREADS
+#    define THREAD_FORMAT "%p:"
+#    define THREAD_ARGUMENT aTHX_
+#  else
+#    define THREAD_FORMAT
+#    define THREAD_ARGUMENT
+#  endif
```
Comment (bram-perl) [inline locale.c:6527]: I wonder if this should be followed by a `#undef THREAD_FORMAT`, `#undef THREAD_ARGUMENT`..

That is
```
# ifdef ...
#   define THREAD_FORMAT ...
# endif

    return ....

# undef THREAD_FORMAT
# undef THREAD_ARGUMENT
```

Right now it's leaving these defined; I don't know what the recommended practice is tho..

File: locale.c [hunk 3]
```diff
@@ -6471,56 +6473,66 @@ Perl_sync_locale()
...
-    static char ret[1024];
-    assert(cat_index <= NOMINAL_LC_ALL_INDEX);
+    const char * locale_quote;
+    const char * retval_quote;
 
-    my_strlcpy(ret, "setlocale(", sizeof(ret));
-    my_strlcat(ret, category_names[cat_index], sizeof(ret));
-    my_strlcat(ret, ", ", sizeof(ret));
+    assert(cat_index <= NOMINAL_LC_ALL_INDEX);
+    PERL_UNUSED_ARG(line);  /* Currently unused */
```
Comment (bram-perl) [inline locale.c]: That doesn't seem to be true?
`line` is used on line 6525

Comment (khwilliamson) [reply locale.c]: You're right

File: locale.c (hunk 1 above)
Comment (khwilliamson) [reply locale.c]: rebase -i error on my part
